pkg/api/persistentvolumeclaim/util.go

```diff
@@ -60,14 +60,18 @@ func dataSourceInUse(oldPVCSpec *core.PersistentVolumeClaimSpec) bool {
 
 func dataSourceIsEnabled(pvcSpec *core.PersistentVolumeClaimSpec) bool {
 	if pvcSpec.DataSource != nil {
+		apiGroup := ""
+		if pvcSpec.DataSource.APIGroup != nil {
+			apiGroup = *pvcSpec.DataSource.APIGroup
+		}
 		if pvcSpec.DataSource.Kind == pvc &&
-			*pvcSpec.DataSource.APIGroup == "" &&
+			apiGroup == "" &&
 			utilfeature.DefaultFeatureGate.Enabled(features.VolumePVCDataSource) {
```

Comment:
I have a slight preference to check the feature gate first.  That way in non-alpha clusters, the nil deref would have been skipped.

Reply:
Good catch, updated.